tfhe/src/core_crypto/algorithms/lwe_encryption.rs

```diff
@@ -294,15 +389,15 @@ pub fn trivially_encrypt_lwe_ciphertext<Scalar, OutputCont>(
     output.get_mut_mask().as_mut().fill(Scalar::ZERO);
 
     let output_body = output.get_mut_body();
-
-    *output_body.data = encoded.0;
-
     let ciphertext_modulus = output_body.ciphertext_modulus();
-    assert!(ciphertext_modulus.is_compatible_with_native_modulus());
-    if !ciphertext_modulus.is_native_modulus() {
-        *output_body.data = (*output_body.data)
-            .wrapping_mul(ciphertext_modulus.get_power_of_two_scaling_to_native_torus());
-    }
+
+    *output_body.data = match ciphertext_modulus.kind() {
+        CiphertextModulusKind::Native | CiphertextModulusKind::NonNative => encoded.0,
+        CiphertextModulusKind::NonNativePowerOfTwo => {
+            // Manage non native power of 2 encoding
+            encoded.0 * ciphertext_modulus.get_power_of_two_scaling_to_native_torus()
+        }
+    };
```

Comment:
Does not look equivalent to what was done before, should it not be ?
```rust 
       CiphertextModulusKind::Native  => encoded.0,
        CiphertextModulusKind::NonNativePowerOfTwo | CiphertextModulusKind::NonNative => {
            // Manage non native power of 2 encoding
            encoded.0 * ciphertext_modulus.get_power_of_two_scaling_to_native_torus()
        }
```

Reply:
That is because before this pr we only had

native
Non native power of 2

So !native modulus == power 2 before, now we have to handle an additional case and only the power of 2 case has a special handling